### PR TITLE
[skipruntime] Garbage collector and subscribe callbacks

### PR DESCRIPTION
**Pull request by skiplabsdaniel**

### CODE REVIEW

**Comment by jberdine** [comment]
@beauby, do you think the compiler test failures look like just stale expected results for compiler errors after your change to add full paths to source locations?

**Comment by beauby** [comment]
> @beauby, do you think the compiler test failures look like just stale expected results for compiler errors after your change to add full paths to source locations?

Yes, exactly.

File: skipruntime-ts/native/src/Extern.sk [hunk 1]
```diff
@@ -507,16 +511,24 @@ native fun notifyOfNotifier(
...
 native fun deleteNotifier(notifier: UInt32): void;
 
 @export("SkipRuntime_createNotifier")
 fun createNotifier(notifier: UInt32): Notifier {
-  Notifier(SKStore.ExternalPointer::create(notifier, deleteNotifier))
+  ExternNotifier(SKStore.ExternalPointer::create(notifier, deleteNotifier))
 }
 
-class Notifier(eptr: SKStore.ExternalPointer) {
+class ExternNotifier(eptr: SKStore.ExternalPointer) extends Notifier {
```
Comment (beauby) [inline skipruntime-ts/native/src/Extern.sk:527]:
```suggestion
class ExternalNotifier(eptr: SKStore.ExternalPointer) extends Notifier {
```

File: skipruntime-ts/native/src/Runtime.sk [hunk 2]
```diff
@@ -453,26 +466,31 @@ fun initService(service: Service): Result<void, .Exception> {
...
       .getPersistent(kRemoteSpecifiers)
       .map(RemoteSpecifiers::type)
       .map(v -> v.value)
-      .fromSome();
-    remoteCollections.each((_k, es) -> es.shutdown());
+      .each(remoteCollections -> {
+        remoteCollections.each((_k, es) -> es.shutdown());
+      });
     context.removePersistent(kRemoteSpecifiers);
+    context.update();
```
Comment (beauby) [inline skipruntime-ts/native/src/Runtime.sk:493]: Was this update missing before?

File: skipruntime-ts/native/src/Runtime.sk [hunk 3]
```diff
@@ -1170,6 +1192,7 @@ fun getAll(
   optRequest: ?Request,
 ): GetResult<Values> {
   resourceInstanceId = Ksuid::create().toString();
+  //subscribe
```
Comment (beauby) [inline skipruntime-ts/native/src/Runtime.sk]: ?

File: skipruntime-ts/native/src/Runtime.sk [hunk 4]
```diff
@@ -1227,23 +1250,70 @@ fun getForKey(
...
+): void {
+  context
+    .getPersistent(`subscription.${sid}`)
+    .map(SKStore.IntFile::type) match {
+  | Some(subId) ->
+    session = subId.value;
+    context.sessions.maybeGet(session).each(sub -> {
+      close = sub.cmd match {
+      | SKStore.NWatch(_, _, _, close, _) -> close
+      | _ -> invariant_violation("Not manage session kind")
```
Comment (beauby) [inline skipruntime-ts/native/src/Runtime.sk:1264]: When can this happen?

**Review by beauby — APPROVED** [review]
LGTM

File: skipruntime-ts/native/src/Extern.sk (hunk 1 above)
Comment (skiplabsdaniel) [reply skipruntime-ts/native/src/Extern.sk:527]: I prefer to keep the naming consistent, if it has to be renamed it's better to do it in another PR by renaming all the other `Extern`.

File: skipruntime-ts/native/src/Runtime.sk (hunk 4 above)
Comment (skiplabsdaniel) [reply skipruntime-ts/native/src/Runtime.sk:1265]: In theory, never.

Comment (skiplabsdaniel) [reply skipruntime-ts/native/src/Runtime.sk:1264]: In theory, never

File: skipruntime-ts/native/src/Runtime.sk (hunk 2 above)
Comment (skiplabsdaniel) [reply skipruntime-ts/native/src/Runtime.sk:493]: Indeed

**Pull request merged**